### DISCUSSION TITLE
Integration of Tuya Local with Home Assistant for Vacuum Mapping Capabilities

### DISCUSSION THREAD

**Discussion by mariustt** [make-all/tuya-local#1382] [General]
Hi everyone,

I am exploring the possibility of integrating a vacuum device with Home Assistant using the Tuya Local library. My main goal is to achieve map visualization within Home Assistant.

I've discovered a library named "Tuya Cloud Vacuum Map Extractor" (https://github.com/oven-lab/tuya_cloud_map_extractor), which seems to export the map created by the vacuum and then import it into Home Assistant.

Has anyone successfully used the Tuya Local integration to display maps in Home Assistant and send task-specific commands to the vacuum, similar to the functionality available in the Tuya Cloud app?

I understand that the native Tuya Local integration may not support mapping features. Therefore, I'm reaching out to see if anyone has found a workaround or solution to integrate vacuum mapping with Home Assistant using local integration.

Any insights, experiences, or suggestions you can share would be greatly appreciated. Looking forward to learning from your experiences and solutions!

Thank you!

**Comment by make-all** [comment]
The maps are generated in the cloud, so it is not possible for a local integration to display maps.

**Comment by mariustt** [comment]
Thank you for your reply @make-all!

Yes, I understand that the maps are generated in the cloud and cannot be generated locally. But my thought would be:
1. Using the tuya app/smart life, a map is generated and exported via API (I guess it would include one or more files)
2. Place the exported file/files in the tuya local files e.g. `custom_components/tuya_local/vacuum_maps`
3. In the device yaml file, specify the path to the specific map e.g. `local_map: /./vacuum_maps/{vacum_model}.extension`
4. For the map mapping, you can probably use some already created homeassistant map e.g. https://github.com/PiotrMachowski/Lovelace-Xiaomi-Vacuum-Map-card
5. Communication between the map and the vacuum is probably based on coordinates that are likely to be received locally by the device

As for the first point 1, it could probably even be automated using the simplest homeassistant RESTful - https://www.home-assistant.io/integrations/rest/ to both save and update the map locally.

Do you think such a plan is not possible? I think it would certainly extend the functionality and user interest of tuya local, as vacuum is probably the one most homeassistants use.
